### PR TITLE
Boost: Update getting started flow

### PR DESCRIPTION
**Pull request by haqadn**
Fixes Automattic/boost-cloud#182

## Proposed changes:
- [x] Connection screen is removed, only getting started page is shown
- [x] Site connection is done automatically when user chooses a plan from getting started page
- [x] If a site connection already exists, you still should see a getting-started page, but choosing a plan shouldn't reinitiate a connection
- [x] If there is any error while establishing site connection, show a snackbar
- [x] Fix e2e tests to work with the updates (currently failing at trying to [establish connection via CLI](https://github.com/Automattic/jetpack/actions/runs/4192667237/jobs/7270837523#step:11:87))

### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

None

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

* You should see the getting started page with pricing table
* Make sure the proposed changes are working
* Try combinations of: free plan, paid plan, jetpack already installed and connected, without preexisting connection)

### CODE REVIEW

File: projects/plugins/boost/app/assets/src/js/pages/getting-started/GettingStarted.svelte [hunk 1]
```diff
@@ -12,19 +15,84 @@
 
 	// svelte-ignore unused-export-let - Ignored values supplied by svelte-navigator.
 	export let navigate, location;
+
+	const initiatingFreePlan = writable( false );
+	const initiatingPaidPlan = writable( false );
+	const dismissedSnackbar = writable( false );
```
Comment (thingalon) [inline projects/plugins/boost/app/assets/src/js/pages/getting-started/GettingStarted.svelte]: Is there a reason to use writable stores here, instead of just `let` variables? They will still update the markup below when changed.

File: projects/plugins/boost/app/assets/src/js/pages/getting-started/GettingStarted.svelte [hunk 2]
```diff
@@ -12,19 +15,84 @@
...
+				if ( $connection.error instanceof ApiError ) {
+					return $connection.error.getDisplayBody();
+				}
+
+				return $connection.error.message;
+			}
+
+			return null;
+		}
+	);
```
Comment (thingalon) [inline projects/plugins/boost/app/assets/src/js/pages/getting-started/GettingStarted.svelte]: This seems like way too much ApiError-specific logic for the header of an unrelated svelte file. Also, I don't think you want `getDisplayBody` here - I think you just want `.message`. DisplayBody JSON-stringifies it.

**Review by thingalon — CHANGES_REQUESTED** [review]
Wow, this makes the connection flow feel so effortless! Nice work!

I have noticed a couple of hiccups:

The snackbar error contains raw JSON when connecting fails, and appears behind parts of the pricing table:
image

If you:
1. Buy Jetpack Boost,
2. Deactivate and reactivate Boost,
3. It then shows you the same "Getting Started" page with two options.
  - Clicking "Start for free" takes you to your Boost dashboard with the premium version you have already purchased, but
  - Clicking on "Get Boost" takes you to a broken cart page on WordPress.com:
image

I understand that we can't show users their purchases until they have connected. But when you click "Get Boost", it shouldn't break.

Is there a way we can detect a prior purchase, or a way that we can improve this bit of the flow for returning users?

File: projects/plugins/boost/app/assets/src/js/pages/getting-started/GettingStarted.svelte (hunk 2 above)
Comment (haqadn) [reply projects/plugins/boost/app/assets/src/js/pages/getting-started/GettingStarted.svelte]: I have removed the usage of `getDisplayBody`. However, I didn't fully get what you mean by `snackbarMessage` being in an unrelated file and what would be the alternative. But, I have also rearranged where the Snackbar component is used.

File: projects/plugins/boost/tests/e2e/lib/env/prerequisites.js [hunk 3]
```diff
@@ -142,19 +126,16 @@ export async function ensureConnectedState( requiredConnected = undefined, page
...
-	// as a localhost site. The only solution is to do it via the site itself running under the localtunnel.
 	const jetpackBoostPage = await JetpackBoostPage.visit( page );
-	await jetpackBoostPage.connect();
+	await jetpackBoostPage.chooseFreePlan();
 	await jetpackBoostPage.isOverallScoreHeaderShown();
 }
 
 export async function disconnect() {
 	logger.prerequisites( `Disconnecting Boost plugin to WP.com` );
-	const cliCmd = 'jetpack-boost connection deactivate';
```
Comment (haqadn) [inline projects/plugins/boost/tests/e2e/lib/env/prerequisites.js:155]: Notice this in your review @thingalon, The reason I am doing this is because, if there are other jetpack plugins installed, boosts cli commands aborts the disconnect, which `jetpack disconnect blog` forces the disconnect.

**Review by thingalon — DISMISSED** [review]
Snackbar is now delicious. 
image

Great work @haqadn!

**Review by thingalon — APPROVED** [review]
